crates/starknet-devnet/src/cli.rs

```diff
@@ -237,19 +244,55 @@ impl Args {
                 url: self.fork_network.clone(),
                 block_number: self.fork_block,
             },
-            disable_account_impersonation: self.disable_account_impersonation,
         };
 
         let RequestResponseLogging { log_request, log_response } =
             RequestResponseLogging::from_rust_log_environment_variable();
 
+        // if restricted_methods are not specified, use default ones
+        let restricted_methods = self.restricted_methods.as_ref().map(|methods| {
+            if methods.is_empty() {
+                DEFAULT_RESTRICTED_JSON_RPC_METHODS
+                    .iter()
+                    .map(|s| s.to_string())
+                    .collect::<Vec<String>>()
+            } else {
+                // remove leading slashes
+                methods
+                    .iter()
+                    .map(|s| s.trim_start_matches('/').to_string())
```

Comment:
In multiple places we are removing this leading slash, isn't a single place enough?

Reply:
no, because 1 place is modyfing the user input, the other is removing the slash from the uri path of the current request. The other is used to construct a collection of default restricted methods

Reply:
There is also an occurrence in `is_uri_path_restricted`, but maybe it's a good thing.

Reply:
it is to check if the uri path is restricted

Reply:
Ah you mean in the middleware, didn't realize it's used there. Resolving.